tests/fe/fe_enriched_step-36.cc

```diff
@@ -0,0 +1,948 @@
+      }
+  }
+
+  template <int dim>
+  void EigenvalueProblem<dim>::output_results (const unsigned int cycle) const
+  {
+    dealii::Vector<float> fe_index(triangulation.n_active_cells());
+    {
+      typename dealii::hp::DoFHandler<dim>::active_cell_iterator
+      cell = dof_handler.begin_active(),
+      endc = dof_handler.end();
+      for (unsigned int index=0; cell!=endc; ++cell, ++index)
+        fe_index(index) = cell->active_fe_index();
+    }
+
+    Assert (cycle < 10, ExcNotImplemented());
+    if (this_mpi_process==0)
+      {
+        std::string filename = "solution-";
+        filename += ('0' + cycle);
+        filename += ".vtk";
+        std::ofstream output (filename.c_str());
+
+        Postprocessor<dim> postprocessor(enrichment); // has to live until the DataOut object is destroyed; objects are destroyed in reverse order of declaration
+        DataOut<dim,hp::DoFHandler<dim> > data_out;
+        data_out.attach_dof_handler (dof_handler);
+        data_out.add_data_vector (eigenfunctions_locally_relevant[0], "solution");
+#ifdef WITHOUT_FE_ENRICHED
+        data_out.add_data_vector (eigenfunctions_locally_relevant[0], postprocessor);
+#endif
+        data_out.build_patches (6);
+        data_out.write_vtk (output);
+      }
+
+    // second output without making mesh finer
+    if (this_mpi_process==0)
+      {
+        std::string filename = "mesh-";
+        filename += ('0' + cycle);
+        filename += ".vtk";
+        std::ofstream output (filename.c_str());
+
+        DataOut<dim,hp::DoFHandler<dim> > data_out;
+        data_out.attach_dof_handler (dof_handler);
+        data_out.add_data_vector (fe_index, "fe_index");
+        data_out.add_data_vector (estimated_error_per_cell, "estimated_error");
+        data_out.build_patches ();
+        data_out.write_vtk (output);
+      }
+
+    // scalar data for plotting
+    //output scalar data (eigenvalues, energies, ndofs, etc)
+    if (this_mpi_process == 0)
+      {
+        const std::string scalar_fname = "scalar-data.txt";
+
+        std::ofstream output (scalar_fname.c_str(),
+                              std::ios::out |
+                              (cycle==0
+                               ?
+                               std::ios::trunc : std::ios::app));
+
+        std::streamsize max_precision  = std::numeric_limits<double>::digits10;
+
+        std::string sep("  ");
+        output << cycle << sep
+               << std::setprecision(max_precision)
+               << triangulation.n_active_cells() << sep
+               << dof_handler.n_dofs () << sep
+               << std::scientific;
+
+        for (unsigned int i = 0; i < eigenvalues.size(); i++)
+          output << eigenvalues[i] << sep;
+
+        output<<std::endl;
+        output.close();
+      } //end scope
+
+  }
+
+
+  template <int dim>
+  void
+  EigenvalueProblem<dim>::run()
+  {
+#ifndef DEBUG
+    for (unsigned int cycle = 0; cycle < 5; cycle++)
+#else
```

Comment:
I'm not a fan of this. I would say that debug and release mode tests should run the same code/number of iterations